src/yarp.c

```diff
@@ -6584,10 +6584,10 @@ parse_target(yp_parser_t *parser, yp_node_t *target, yp_token_t *operator, yp_no
 
         if (value != NULL) {
           if (target->as.call_node.arguments == NULL) {
-            target->as.call_node.arguments = yp_arguments_node_create(parser);
+            target->as.call_node.arguments = YP_NODE_DOWNCAST(yp_arguments_node_create(parser));
```

Comment:
Could you create a way in `config.yml` to specify the type? Right now `arguments` on CallNode is

```yaml
      - name: arguments
        type: node?
```

what do you think of

```yaml
      - name: arguments
        type: node?
        kind: ArgumentsNode
```

In which case it would template out a `yp_arguments_node_t *` on the AST.